### ISSUE TITLE
/data/data folder fails to list all subfolders from v3.6.6

### ISSUE THREAD

**Issue by NobodyNext** [TeamAmaze/AmazeFileManager#3033]
**Describe the bug**
From version 3.6.6, `/data/data` folder only lists few app data folders (with root access). No error message or anything, it simply fails to lists all appdata folders inside `/data/data`.  It was working properly until version 3.6.5

**To Reproduce**
1. Go to '/data/data/' folder

**Expected behavior**
Should lists all subfolders since it has root access

**Smartphone (please complete the following information):**
 - Device: Redmi K30
 - OS: 11
 - Rooted: Yes
 - Version: 3.6.7

**Comment by VishalNehra** [comment]
@AsyncRes @EmmanuelMess

**Comment by EmmanuelMess** [comment]
What properties do those folders have? Are those folders simlinks?

**Comment by AsyncRes** [comment]
> What properties do those folders have? Are those folders simlinks?

Are you talking about subfolders of /data/data? 
Only two folders are listed, `com.amaze.filemanager` and `com.google.android.gms`.

Normally all apps data folders has has `drwx------` permission. There is only one exception, `com.google.android.gms` has `drwx--x--x`. 

That's why amaze only has access to list it's own data folder, and the `gms` folder.
But I'm accessing /data/data using root access. So, why it's not showing all folders? This problem started since 3.6.6

**Comment by EmmanuelMess** [comment]
I think this function should be able to return an error state:

https://github.com/TeamAmaze/AmazeFileManager/blob/bdf71e99028ccb2ccdb1beb36c143440aeddaafb/file_operations/src/main/c/rootoperations.c#L18-L25

**Comment by AsyncRes** [comment]
No error dialog. 
I checked some other root folders, seems many folders are missing. 
For example /data/app/[random-str]/ is showing empty, while it actually has a subfolder which has `drwxrwxr-x` permission, still it failed to list. So, clearly something is wrong in the code despite having access permission.

**Comment by EmmanuelMess** [comment]
Should it throw a java exception from native?

**Comment by AsyncRes** [comment]
Sorry I have no idea, I'm just explaining the problem.

**Comment by EmmanuelMess** [comment]
> Should it throw a java exception from native?

@VishalNehra @TranceLove

**Comment by TranceLove** [comment]
> Should it throw a java exception from native?

I'm in favour of this, for Java/Kotlin code to give feedback to user.

**Comment by PragmaTwice** [comment]
Any progress here?

**Comment by VishalNehra** [comment]
> Any progress here?
See #3171 
btw i see ur S;G fan ;)

**Comment by TranceLove** [comment]
@PragmaTwice I can only say, I was able to reproduce the problem on an rooted Android 12 emulator.

I thought it was the libsuperuser, so I tried using the more modern libsu (see #3171), but still no improvement.

In debugging session, I saw both code and ADB shell (su-ed) are calling the same busybox/toybox, but two sides gave two results - with code side only yields two folders while ADB shell gave all folders.

So I think something might be going on beneath.

May try using embedded busybox as libsu provides, but don't expect too much😞
